packages/allow-scripts/src/config.js

```diff
@@ -36,9 +37,10 @@ const bannedBins = new Set(['corepack', 'node', 'npm', 'pnpm', 'yarn'])
 /**
  * @param {Object} args
  * @param {string} args.rootDir
+ * @param {string[]=} args.lifecycleEvents - which script names to consider
```

Comment:
```js
/**
 * @param {string[]=} args.lifecycleEvents - which script names to consider
 */
```

```js
/**
 * @param {string[]} [args.lifecycleEvents] - which script names to consider
 */
```

```js
/**
 * @param {string[]} args.lifecycleEvents - which script names to consider
 */
async function loadAllPackageConfigurations({ rootDir, lifecycleEvents = DEFAULT_LIFECYCLE_EVENTS }) {
```


All three of these express the same thing. In the third, the default assignment _implies_ that `lifecycleEvents` is optional. The default assignment takes precedence over the first two; any combination of the third with the first or second is superfluous from the compiler's perspective.

We should probably try to pick a style.  As far as enforcing the style, we could adopt [eslint-plugin-jsdoc](https://npm.im/eslint-plugin-jsdoc) _but_ it may conflict w/ `prettier-plugin-jsdoc` and cause a headache.

Anyway; I don't really care which we use as long as it's consistent, but it's currently inconsistent. Low priority, just thought I'd mention

Reply:
> All three of these express the same thing. In the third, the default assignment implies that lifecycleEvents is optional.

Fake news.  

<details>
https://github.com/legobeat/LavaMoat/pull/19
```
> lavamoat-monorepo@0.0.0 build:types
> tsc -b

packages/allow-scripts/src/report.js:26:42 - error TS2345: Argument of type '{ rootDir: string; }' is not assignable to parameter of type '{ rootDir: string; lifecycleEvents: string[]; }'.
  Property 'lifecycleEvents' is missing in type '{ rootDir: string; }' but required in type '{ rootDir: string; lifecycleEvents: string[]; }'.

26   } = await loadAllPackageConfigurations({ rootDir })
                                            ~~~~~~~~~~~

  packages/allow-scripts/src/config.js:40:4
    40  * @param {string[]} args.lifecycleEvents - which script names to consider
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    41  * @returns {Promise<PkgConfs>}
       ~~~
    'lifecycleEvents' is declared here.

packages/allow-scripts/src/runAllowedPackages.js:26:42 - error TS2345: Argument of type '{ rootDir: string; }' is not assignable to parameter of type '{ rootDir: string; lifecycleEvents: string[]; }'.
  Property 'lifecycleEvents' is missing in type '{ rootDir: string; }' but required in type '{ rootDir: string; lifecycleEvents: string[]; }'.

26   } = await loadAllPackageConfigurations({ rootDir })
                                            ~~~~~~~~~~~

  packages/allow-scripts/src/config.js:40:4
    40  * @param {string[]} args.lifecycleEvents - which script names to consider
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    41  * @returns {Promise<PkgConfs>}
       ~~~
    'lifecycleEvents' is declared here.


Found 2 errors.
```
</details>

- In any case, addressing type consistency in #1381